src/Api/PubnubApi/EventEngine/EffectDispatcher.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace PubnubApi.PubnubEventEngine
+{
+	public class EffectDispatcher
+	{
+		public enum DispatcherType
```

Comment:
Are we going to need this? I mean `disptach` method will only care of invocations.